### PR TITLE
Refactor Navigation Bar: Add Menu Folders for Improved Organization and Include CTA

### PR DESCRIPTION
**Pull request by kccrtv**
# Description and Motivation
This pull request refactors the existing navigation bar by introducing menu folders (dropdowns) to better organize the menu options and enhance user navigation. The restructured menu allows for grouping related links under expandable sections, improving the overall user experience by reducing visual clutter and providing a more intuitive navigation structure. Additionally, addresses issue #1127 

In addition to the menu folders, a Call to Action (CTA) has been added to the navigation bar, providing users with quick access to our data landing page. This enhancement aims to increase user engagement by making key actions more prominent and accessible.

## Has this been tested? How?
Tests passing

## Screenshots (if appropriate)

> ### Previous version at 1279px
> Screenshot 2024-08-26 at 9 43 27 PM

> ### New version at 1279px
> Screenshot 2024-08-26 at 9 41 42 PM
> 

> ### Previous version at 969px to 768px
> Screenshot 2024-08-26 at 9 43 45 PM
> Screenshot 2024-08-26 at 9 44 19 PM

> ### New version at 969px
> Screenshot 2024-08-26 at 9 42 00 PM

> ### Previous version - mobile
> Screenshot 2024-08-26 at 9 44 37 PM

> ### New version - mobile
> Screenshot 2024-08-26 at 9 42 26 PM

> ### New version - desktop folders

> Screenshot 2024-08-26 at 9 40 50 PM
> Screenshot 2024-08-26 at 9 40 56 PM
> Screenshot 2024-08-26 at 9 41 01 PM

## Types of changes
- New content or feature
- Refactor / chore

### CODE REVIEW

File: frontend/src/utils/urlutils.tsx [hunk 1]
```diff
@@ -93,6 +96,34 @@ export const ADDED_MOBILE_PAGE_URL_TO_NAMES: Record<string, string> = {
   '/': 'Home',
 }
 
+export const NAVIGATION_STRUCTURE = {
+  about: {
+    label: 'About',
+    pages: {
+      [WHAT_IS_HEALTH_EQUITY_PAGE_LINK]: 'What is Health Equity?',
+      [GUN_VIOLENCE_POLICY]: 'Policy Context',
```
Comment (eriwarr) [inline frontend/src/utils/urlutils.tsx:104]: Should we comment the policy context option out for now until the page is up?

**Review by eriwarr — COMMENTED** [review]
This looks awesome!! Just made a few suggestions.

File: frontend/src/utils/urlutils.tsx (hunk 1 above)
Comment (kccrtv) [reply frontend/src/utils/urlutils.tsx:104]: sure!

**Review by eriwarr — APPROVED** [review]
Awesome work!

**Pull request merged**